assign-doi-datacite/src/main/java/no/unit/nva/datacite/mdsclient/DataciteMdsConnectionFactory.java

```diff
@@ -0,0 +1,97 @@
+ * during server challenge (pre-emptive authentication).
+ *
+ * <p>Use {@link #getAuthenticatedConnection(String)} to construct a new authenticated API connection.
+ *
+ * @see #createNvaCustomerAuthenticator(String)
+ */
+public class DataciteMdsConnectionFactory {
+
+    public static final PasswordAuthentication DO_NOT_SEND_CREDENTIALS = null;
+    private final PasswordAuthenticationFactory authenticationFactory;
+    private final String mdsHostname;
+    private final int mdsPort;
+    private final Builder httpBuilder;
+
+    /**
+     * Default constructor.
+     *
+     * @param authenticationFactory Authentication factory which is used for 401 challenge responses.
+     * @param mdsHostname           MDS API hostname
+     * @param mdsPort               MDS API port
+     */
+    public DataciteMdsConnectionFactory(PasswordAuthenticationFactory authenticationFactory,
+                                        String mdsHostname,
+                                        int mdsPort) {
+        this(HttpClient.newBuilder(), authenticationFactory, mdsHostname, mdsPort);
+    }
+
+    /**
+     * Constructor for testing.
+     *
+     * @param httpBuilder           HttpClient to override security configuration
+     * @param authenticationFactory Authentication factory which is used for 401 challenge responses.
+     * @param mdsHostname           MDS API hostname
+     * @param mdsPort               MDS API port
+     */
+    public DataciteMdsConnectionFactory(HttpClient.Builder httpBuilder,
+                                        PasswordAuthenticationFactory authenticationFactory,
+                                        String mdsHostname,
+                                        int mdsPort) {
+        this.authenticationFactory = authenticationFactory;
+        this.mdsHostname = mdsHostname;
+        this.mdsPort = mdsPort;
+        this.httpBuilder = httpBuilder;
+    }
+
+    public DataCiteMdsConnection getAuthenticatedConnection(String customerId) {
+        return new DataCiteMdsConnection(httpBuilder
+            .version(Version.HTTP_2)
+            .connectTimeout(Duration.ofSeconds(2))
+            .authenticator(createNvaCustomerAuthenticator(customerId))
+            .build(), mdsHostname, mdsPort);
+    }
+
+    private Authenticator createNvaCustomerAuthenticator(String customerId) {
+        return new Authenticator() {
+
+            @Override
+            public PasswordAuthentication requestPasswordAuthenticationInstance(String host, InetAddress addr,
+                                                                                int port, String protocol,
+                                                                                String prompt, String scheme,
+                                                                                URL url, RequestorType reqType) {
+                if (host.equalsIgnoreCase(mdsHostname) && port == mdsPort) {
```

Comment:
Clean up this block?

Reply:
Good catch, its conditional check is now:

 ```
if (isCommunicatingTowardsConfiguredDataciteApi(host, port)) ...
```